src/lib.rs

```diff
@@ -357,12 +362,21 @@ impl ThreadPool {
     ///
     /// # Panics
     ///
-    /// This function will panic if max_size is 0 or lower than core_size.
-    pub fn new_named(name: String, core_size: u32, max_size: u32, keep_alive: Duration) -> Self {
+    /// This function will panic if max_size is 0, lower than core_size or exceeds half
+    /// the size of usize. This restriction exists because two counters (total workers and
+    /// idle counters) are stored within one AtomicUsize.
+    pub fn new_named(
+        name: String,
+        core_size: usize,
+        max_size: usize,
+        keep_alive: Duration,
+    ) -> Self {
         let (sender, receiver) = crossbeam_channel::unbounded();
 
         if max_size == 0 || max_size < core_size {
             panic!("max_size must be greater than 0 and greater or equal to the core pool size");
+        } else if max_size > MAX_SIZE {
+            panic!("max_size may not be larger than half the size of usize");
```

Comment:
".. than half the size of usize" is probably not obvious for someone without context, who would probably think of it as `usize::MAX / 2`.

This is hard to word perfectly, but maybe ".. than usize with the lower half of bits set"

Reply:
Agreed, I've improved the error message to clear up the ambiguity.